subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginContainer.java

```diff
@@ -103,9 +103,16 @@ public boolean hasPlugin(Class<? extends Plugin> type) {
     private Plugin doFindPlugin(String id) {
         for (final PluginManagerInternal.PluginWithId pluginWithId : pluginManager.pluginsForId(id)) {
             Plugin plugin = Iterables.tryFind(DefaultPluginContainer.this, new Predicate<Plugin>() {
+                @Override
                 public boolean apply(Plugin plugin) {
                     return pluginWithId.clazz.equals(plugin.getClass());
                 }
+
+                @Override
+                // Added for Java 6 source compatibility
+                public boolean test(Plugin input) {
+                    return apply(input);
+                }
```

Comment:
Couldn't all of these `// Added for Java 6 source compatibility` simply be fixed by creating a `CompatiblePredicate` that implemented this override the same way as the `Predicate` default method?

That way you don't need the duplicate override ever place you need to support Java 6 compatibility?